desc/compute/utils.py

```diff
@@ -91,6 +92,27 @@ def compute(parameterization, names, params, transforms, profiles, data=None, **
         data=data,
         **kwargs,
     )
+
+    # convert data from default 'rpz' basis to 'xyz' basis, if requested by the user
+    parameterization = _parse_parameterization(parameterization)
+    for name in data.keys():
+        if (
+            data_index[parameterization][name]["dim"] == 3  # vector quantity
+            and kwargs.get("basis", "rpz").lower() == "xyz"  # xyz basis requested
+        ):
+            from .geom_utils import rpz2xyz, rpz2xyz_vec
+
+            if name == "x":
+                data[name] = rpz2xyz(data[name])
+            else:
+                errorif(
+                    "phi" not in data.keys(),
+                    ValueError,
+                    "'phi' must be included in the compute data "
+                    + "to convert to 'xyz' basis.",
+                )
```

Comment:
It would be convenient if we could automatically add `"phi"` to the list of compute quantity `names` when `basis="xyz"` is requested. Unfortunately I think that logic would have to be added to each class's `compute` method, since that is typically where we get the params that are needed for the list of compute quantities.

Reply:
In some curve functions, this was previously done by using transforms. So, I had a different if statement for `phi` is not included in the `data`. But yeah just having `phi` always should help.